src/directives/formly-form.test.js

```diff
@@ -933,12 +933,19 @@ describe('formly-form', () => {
 
       const listener = sinon.spy()
       const expression = sinon.spy()
+      const deepLinkField = getNewField()
+      deepLinkField.key = 'foo.bar'
+      deepLinkField.watcher = {
+        listener,
+        expression: sinon.spy(),
```

Comment:
Does this do the trick?

it(`should add watches on deep dive fields`, () => {
      debugger
      const formWithOptions = '<formly-form model="model" fields="fields" options="options"></formly-form>'
      scope.model = {}
      scope.options = {}

      const deepLinkField = getNewField()
      deepLinkField.key = 'foo.bar'
      deepLinkField.watcher = {
        listener: sinon.spy(),
      }

      scope.fields = [deepLinkField]
      compileAndDigest(formWithOptions)
      expect(compileAndDigest).to.not.throw()
      expect(deepLinkField.watcher.listener).to.have.been.called
      scope.model[deepLinkField.key] = 'brown'
      scope.$digest()
      expect(compileAndDigest).to.not.throw()
      expect(deepLinkField.watcher.listener).to.have.been.called
    })

From:  "Kent C. Dodds" <notifications@github.com>
Reply-To:  formly-js/angular-formly <reply@reply.github.com>
Date:  Friday, November 13, 2015 at 9:52 AM
To:  formly-js/angular-formly <angular-formly@noreply.github.com>
Cc:  Matt Kruczek <kruz79@gmail.com>
Subject:  Re: [angular-formly] fix(formly-form): Issue number 542 (#547)


In src/directives/formly-form.test.js:
> @@ -933,12 +933,19 @@ describe('formly-form', () => {
>  
>        const listener = sinon.spy()
>        const expression = sinon.spy()
> +      const deepLinkField = getNewField()
> +      deepLinkField.key = 'foo.bar'
> +      deepLinkField.watcher = {
> +        listener,
> +        expression: sinon.spy(),
Sorry, I think I told you wrong. The expression shouldn't be here. We want to test behavior for when an expression is omitted. So this should be:

deepLinkField.watcher = {
  listener: sinon.spy(),
  // no expression
}
Then we need to manually change the value of foo.bar, run scope.$digest() and validate that the listener was called with the new value of foo.bar. I suggest that this all be put into its own it statement.

—
Reply to this email directly or view it on GitHub.